test/integration/upstreams/per_host_upstream_config.cc

```diff
@@ -0,0 +1,20 @@
+#include "test/integration/upstreams/per_host_upstream_config.h"
+
+#include "test/integration/upstreams/per_host_upstream_request.h"
+
+namespace Envoy {
+
+Router::GenericConnPoolPtr PerHostGenericConnPoolFactory::createGenericConnPool(
```

Comment:
can we get away with .cc files only for this and below, as we do with test/integration/filters?

Most of the factory-only code doesn't need header files but I can't recall if this does.

Also as I think upstreams/config will be 1:1 I think we can stick them in the same file to make it easier to reason about and have one top-of-file comment of what it's for.

Reply:
merge config.h and upstream_request.h